.github/workflows/build_push_img.yml

```diff
@@ -1,7 +1,36 @@
 name: Build and Push Marqo Docker Image
 
 on:
-  workflow_dispatch:
+  workflow_call:
+    inputs:
+      marqo_ref:
+        description: 'Marqo branch-name, commit SHA or tag'
+        required: false
+        type: string
+        default: 'mainline'
+      push_to:
+        description: 'image destination. Options: "ECR", "DockerHub"'
+        required: true
+        type: string
+        default: 'ECR'
+      image_repo:
+        description: 'Image repository'
+        required: true
+        type: string
+        default: 'marqo'
+      image_tag:
+        description: 'Image tag'
+        required: true
+        type: string
+      dockerhub_username:
+        description: 'DockerHub username'
+        required: false
+        type: string
+      dockerhub_password:
```

Comment:
I wonder if having these as configurable inputs is a security risk. Are the values persisted anywhere? Why don't we read from GH secrets? Alternatively support only ecr which is all we need for this

Reply:
Unless we can verify this isn't a security bad practice

Reply:
I believe this is copied from https://github.com/marqo-ai/open_source_CI/blob/main/.github/workflows/build_push_img.yml. I agree that a better way is to store credentials in action secrets. 
This pipeline could replace the one mentioned above and be used to automate the image release to dockerhub in the future.

Reply:
After discussion with Yihan, removed the dockerhub related code from this github action entirely